sdk/communication/communication-phone-numbers/test/public/utils/msUserAgentPolicy.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  PipelinePolicy,
+  PipelineRequest,
+  PipelineResponse,
+  SendRequest,
+} from "@azure/core-rest-pipeline";
+import { env } from "@azure-tools/test-recorder";
+
+export function createMSUserAgentPolicy(): PipelinePolicy {
+  return {
+    name: "msUserAgentPolicy",
+    async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
+      const useragent = env.AZURE_USERAGENT_OVERRIDE;
+      if (useragent) {
```

Comment:
Considering that this policy is only intended for running tests, I would prefer to throw an Error if the variable is not found, at least when not running in playback mode. This way, we would know if the tests are not configured properly. What are your thoughts on this?

Reply:
this configuration is not needed to run the tests, it is only will come in hand when running this on the test pipeline, where this info will be used to mark traffic as test traffic, for metrics reasons. With that been said I don't see a need for this variable to be required.

Reply:
All right then, also I guess that not requiring it will be consistent with the other languages 👍